src/bin/pgcopydb/ld_test_decoding.c

```diff
@@ -300,6 +302,55 @@ parseTestDecodingMessage(StreamContext *privateContext,
 }
 
 
+/*
+ * findIdentifierEndPos returns the first position of the separator in the given
+ * message after navigating through all the quotes. It assumes that the message
+ * contains a well-formed identifier and doesn't attempt to handle errors.
+ */
+static int
+findIdentifierEndPos(const char *message, char separator)
+{
+	int quoteCount = 0;
+
+	int pos = 0;
+
+	for (pos = 0; message[pos] != '\0'; pos++)
+	{
+		if (message[pos] == '"')
+		{
+			quoteCount++;
+		}
+
+		/*
+		 * We are looking for the first 'separator' in the message which
+		 * should be outside the quotes.
+		 *
+		 * When there is 'separator' inside quotes, quoteCount will be odd,
+		 * and we need to account it as a part of the identifier.
+		 *
+		 * Here are some possible inputs and ^ indicates the position
+		 * we want to find:
+		 *
+		 * separator: '.'
+		 * "Foo.Bar"."Baz": UPDATE:
+		 *          ^
+		 * separator: '.'
+		 * "Foo Bar.Baz".hello: UPDATE:
+		 *		        ^
+		 * separator: ':'
+		 * "Foo Bar.Baz": UPDATE:
+		 *		        ^
+		 */
+		else if (message[pos] == separator && quoteCount % 2 == 0)
+		{
+			break;
```

Comment:
On a first look, I believe this misses the possibility of an identifier embedding quotes using the Postgres syntax `"Foo""Bar"` for `Foo"Bar`.

Reply:
On-top of that we have to take care of unicode special syntax, see https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS

Reply:
Right, I will check on the unicode part, but "Foo""Bar" is working as expected. I modified the test to include "Foo""Bar".

Reply:
@dimitri The current implementation also seem to handle unicode chars, I've also added tests with unicode chars which covers unicode in schema, table name, column and types.

Reply:
Thanks for the added test case! Also I now realize why `"Foo""Bar"` works and my initial reading of your code was wrong. I have added a comment about that, I think we should make sure that the separator can't be `"`.